### PR TITLE
feat (worker, api): spawn infos err prerequisites

### PR DESCRIPTION
**Pull request by yesnault**
Signed-off-by: Yvonnick Esnault <yvonnick.esnault@corp.ovh.com>

### CODE REVIEW

File: engine/api/router.go [hunk 1]
```diff
@@ -40,16 +40,17 @@ type Handler func(w http.ResponseWriter, r *http.Request, db *gorp.DbMap, c *bus
...
+	post                 Handler
+	put                  Handler
+	deleteHandler        Handler
+	auth                 bool
+	isExecution          bool
+	needAdmin            bool
+	needUsernameOrAdmin  bool
+	needHatchery         bool
+	needWorker           bool
+	needHatcheryOrWorker bool
```
Comment (fsamin) [inline engine/api/router.go]: i don't like it

File: engine/api/router.go [hunk 2]
```diff
@@ -373,6 +380,14 @@ func NeedWorker() RouterConfigParam {
 	return f
 }
 
+// NeedHatcheryOrWorker set the route for worker or hatchery only
+func NeedHatcheryOrWorker() RouterConfigParam {
```
Comment (fsamin) [inline engine/api/router.go]: i prefer use needWorker, needHatchery

File: engine/worker/cmd_main.go [hunk 3]
```diff
@@ -222,30 +223,47 @@ func mainCommandRun(w *currentWorker) func(cmd *cobra.Command, args []string) {
...
 					ok, err := checkRequirement(w, r)
-					if err != nil {
-						postCheckRequirementError(&r, err)
-						requirementsOK = false
-						continue
-					}
-					if !ok {
+					if err != nil || !ok {
+						if err != nil {
+							log.Warning("checkQueue> Err on checkRequirement %s", err)
```
Comment (fsamin) [inline engine/worker/cmd_main.go]: Error

Comment (fsamin) [inline engine/worker/cmd_main.go]: if err != nil {
blabla
}

if !ok {

}

File: engine/worker/cmd_main.go [hunk 4]
```diff
@@ -222,30 +223,47 @@ func mainCommandRun(w *currentWorker) func(cmd *cobra.Command, args []string) {
...
-					if j.ID == w.bookedJobID {
-						t = ", this was my booked job"
-					}
 					log.Info("checkQueue> Taking job %d%s", j.ID, t)
 					w.takePipelineBuildJob(ctx, j.ID, j.ID == w.bookedJobID)
 				} else {
-					log.Debug("Unable to run this job, let's continue")
+					log.Debug("Unable to run this job, let's continue %d%s", j.ID, t)
+
+					if j.ID == w.bookedJobID && !firstViewBooked {
```
Comment (fsamin) [inline engine/worker/cmd_main.go]: move this code to line 164 where we manage specifically the bookedJob

File: engine/api/main_routes.go [hunk 5]
```diff
@@ -186,10 +186,9 @@ func (router *Router) init() {
 
 	// Build queue
 	router.Handle("/queue", GET(getQueueHandler))
-	router.Handle("/queue/requirements/errors", POST(requirementsErrorHandler))
```
Comment (fsamin) [inline engine/api/main_routes.go]: Remove the handler

Comment (yesnault) [inline engine/api/main_routes.go]: done

**Pull request merged**